program/src/state/rule_set.rs

```diff
@@ -13,7 +10,7 @@ pub struct RuleSet {
     /// Owner (creator) of the RuleSet.
     owner: Pubkey,
     /// A map to determine the `Rule` that belongs to a given `Operation`.
-    pub operations: HashMap<Operation, Rule>,
+    pub operations: HashMap<u16, Rule>,
```

Comment:
Is there any technical reason we should instead make this a u32 or u64?  For example, in embedded C, if you make a struct with a u16, I think gcc will always still align the struct on a 4- or 8-byte boundary depending on architecture, unless you tell the compiler to pack the structure.  The only reason I ask is because u16 is a strange type, not a byte but not a native int on anything.

Reply:
I believe Rust packs by default. So no real additional space is consumed here. U16 was just because 256 options seems limiting.